File: HW/HW2/hw2.py
#Assuming the API bool isBadVersion is applied
import math

def firstBadVersion(n: int):
    left = 0
    right = 0
    solution = 0

    while left <= right:
        midpoint = left + math.floor((right - left) / 2)
        if isBadVersion(midpoint):
            solution = midpoint
            right = midpoint - 1
        else:
            left = midpoint + 1
    return solution

File: HW/HW2/hw2.java
//Assuming the API bool isBadVersion is applied

public class hw2{
    public int firstBadVersion(int n){
        int left = 0;
        int right = n;
        int solution = 0;

        while (left <= right){
            int midpoint = left + (right-left) / 2;
            if (isBadVersion(midpoint)){
                solution = midpoint;
                right = midpoint - 1;
            }
            else {
                left = midpoint + 1;
            }
        }
        return solution;
    }
}
